improver/wxcode/weather_symbols.py

```diff
@@ -268,95 +267,17 @@ def invert_condition(self, condition):
-            string:
-                The formatted condition statement,
-                e.g.::
-
-                  cubes.extract(Constraint(
-                          name='probability_of_rainfall_rate_above_threshold',
-                          coord_values={'threshold': 0.03})
-                                )[0].data < 0.5)
-        """
-        if isinstance(extract_constraint, list):
-            formatted_str = ""
-
-            for item in extract_constraint:
-                if is_variable(item):
-                    formatted_str += f" cubes.extract({item})[0].data"
-                else:
-                    formatted_str += " " + item
-            return f"({formatted_str}) {condition} {probability_threshold}"
-        return "cubes.extract({})[0].data {} {}".format(
-            extract_constraint, condition, probability_threshold
-        )
-
-    @staticmethod
-    def format_condition_chain(conditions, condition_combination="AND"):
-        """
-        Chain individual condition statements together in a format that
-        numpy.where can use to make a series of comparisons.
-
-        Args:
-            conditions (list):
-                A list of conditions to be combined into a single comparison
-                statement.
-            condition_combination (str):
-                The method by which multiple conditions should be combined,
-                either AND or OR.
-        Returns:
-            string:
-                A string formatted as a chain of conditions suitable for use in
-                a numpy.where statement.
-                e.g. (condition 1) & (condition 2)
-        """
-        combinator = " | " if condition_combination == "OR" else " & "
-        return combinator.join(map("({})".format, conditions))
-
     def create_condition_chain(self, test_conditions):
         """
-        A wrapper to call the construct_condition function for all the
-        conditions specified in a single query.
+        Construct a list of all the conditions specified in a single query.
 
         Args:
             test_conditions (dict):
                 A query from the decision tree.
         Returns:
-            list of str:
-                A list of strings that describe the conditions comprising the
-                query.
-                e.g.::
-
-                  [
-                    "(cubes.extract(Constraint(
-                          name='probability_of_rainfall_rate_above_threshold',
-                          coord_values={'threshold': 0.03})
-                     )[0].data < 0.5) |
-                     (cubes.extract(Constraint(
-                          name=
-                          'probability_of_lwe_snowfall_rate_above_threshold',
-                          coord_values={'threshold': 0.03})
-                     )[0].data < 0.5)"
-                  ]
+            list:
+                A valid condition chain, suitable for passing to 
+                `evaluate_condition_chain`.
```

Comment:
It would be helpful if you could do something similar to what was in the old version, to explain exactly what is returned from this function.  My understanding would be it can return:

- A single constraint, or 
- An expression that is a combination of constraints (?), or
- A two item list with two constraints or expressions in the first item, and a comparison operator describing how to apply them (`AND` `OR` `>` `<` etc) in the second

But this is difficult to infer.  And if I have inferred wrong, even more important to describe it clearly in this docstring!

Reply:
I've now copied the explanation from `evalutate_condition_chain` into this docstring.